src/kinect/updater.cpp

```diff
@@ -111,26 +148,43 @@ std::vector<EntityUpdate> mergeOverlappingXYConvexHulls(const rgbd::Image& image
       const EntityUpdate& u2 = updates[j];
 
       // If we collide, update the i convex hull
-      if (ed::convex_hull::collide(u1.chull, u1.pose_map.t, u2.chull, u2.pose_map.t, 0, 1e6))
+      if (ed::convex_hull::collide(u1.chull, u1.pose_map.t, u2.chull, u2.pose_map.t, 0, 1e6))  // This should prevent multiple entities above each other;1e6 is ok, because objects in other areas are ignored.
+//      if (ed::convex_hull::collide(u1.chull, u1.pose_map.t, u2.chull, u2.pose_map.t, 0, 0.0))  // This way, we get multiple entities above each other
       {
+        ROS_INFO("Collition item %i with %i", i, j);
+        ROS_INFO("Item %i: xyz: %.2f, %.2f, %.2f, z_min: %.2f, z_max: %.2f", i, u1.pose_map.t.getX(), u1.pose_map.t.getY(), u1.pose_map.t.getZ(), u1.chull.z_min, u1.chull.z_max);
+        ROS_INFO("Item %i: xyz: %.2f, %.2f, %.2f, z_min: %.2f, z_max: %.2f", j, u2.pose_map.t.getX(), u2.pose_map.t.getY(), u2.pose_map.t.getZ(), u2.chull.z_min, u2.chull.z_max);
         collission_map[i].push_back(j);
         collided_indices.push_back(j);
       }
     }
   }
 
-  // Now again loop over the updates and only push back in the new updates if it did not collide
+  // Now again loop over the updates and only push back in the new updates if it will not be merged into an other entity
   for (int i = 0; i < updates.size(); ++i)
   {
-    const EntityUpdate& u = updates[i];
-
-    // If index already collided, it will be merged to another one
+    for (std::vector<int>::iterator it = collission_map[i].begin(); it != collission_map[i].end(); ++it)
+    // If index in collided_indices, it will be merged to another one
     if (std::find(collided_indices.begin(), collided_indices.end(), i) == collided_indices.end())
     {
-      // Only push back if not in collided entity
+      // Merging is done by creating a new convexHull. Multiple objects can be merged into one.
+      // No sorting is done. So depending on which entity was taken first, that one is used as basis for merging. But that shouldn't matter, only for UUID.
```

Comment:
So what happens with the UUID? Might it happen that an existing entity switches ID because of this? If so, things go wrong: the entity to grasp is stored in a designator but upon verifying its position it disappears

Reply:
This is before association. So therefore the UUID will be associated to the correct UUID from the previous measurement.